packages/flutter_tools/test/test_compiler_test.dart

```diff
@@ -30,9 +32,24 @@ void main() {
             residentCompiler,
           );
         },
+        overrides: <Type, Generator>{
+          FileSystem: () {
+            const LocalFileSystem fs = LocalFileSystem();
+            tmpDir = fs.systemTempDirectory.createTempSync('flutter_test_compiler_test.');
+            originalCurrentDirectory = fs.currentDirectory;
+            fs.currentDirectory = tmpDir;
+            return fs;
+          }
+        },
       );
     });
 
+    tearDown(() {
+      tryToDelete(tmpDir);
+      testCompiler.dispose();
+      fs.currentDirectory = originalCurrentDirectory;
+    });
+
```

Comment:
Can we add some test cases to verify that lockfile is created successfully?

Reply:
Done.